src/Components/Navbar/Navbar.tsx

```diff
@@ -1,73 +1,83 @@
+/** @format */
 
-import MenuIcon from '@mui/icons-material/Menu';
-import AppBar from '@mui/material/AppBar';
-import Avatar from '@mui/material/Avatar';
-import Box from '@mui/material/Box';
-import Container from '@mui/material/Container';
-import IconButton from '@mui/material/IconButton';
-import Toolbar from '@mui/material/Toolbar';
-import Tooltip from '@mui/material/Tooltip';
-import Typography from '@mui/material/Typography';
-import './Navbar.module.scss';
-import { useDispatch } from 'react-redux';
-import { useEffect, useState } from 'react';
-import { UserDetails } from '@/Redux/Features/Users/GetUserDetailsSlice';
-import { defaultImage } from '@/Assets/Images';
+import MenuIcon from "@mui/icons-material/Menu";
+import AppBar from "@mui/material/AppBar";
+import Avatar from "@mui/material/Avatar";
+import Box from "@mui/material/Box";
+import Container from "@mui/material/Container";
+import IconButton from "@mui/material/IconButton";
+import Toolbar from "@mui/material/Toolbar";
+import Tooltip from "@mui/material/Tooltip";
+import Typography from "@mui/material/Typography";
+import "./Navbar.module.scss";
+import { useDispatch } from "react-redux";
+import { useEffect, useState } from "react";
+import { UserDetails } from "@/Redux/Features/Users/GetUserDetailsSlice";
+import { defaultImage } from "@/Assets/Images";
 
 const Navbar = ({ showDrawer }: any) => {
-
   const dispatch = useDispatch();
-  const [userData, setUserData] = useState()
-  const userId = localStorage.getItem("userId")
+  const [userData, setUserData] = useState();
+  const userId = localStorage.getItem("userId");
   useEffect(() => {
-    getData()
-  }, []);
+    getData();
+  }, [setUserData]);
 
   const getData = async () => {
```

Comment:
You can move this function to the useEffect to prvent the warning of needing to add it to the dependency array. You can also remove `setUserData` from dependency array as it is already managed by React.